pkg/loop/internal/relayer/pluginprovider/chainreader/chain_reader.go

```diff
@@ -330,37 +396,53 @@ func newPbBatchGetLatestValuesReply(result types.BatchGetLatestValuesResult, enc
 				return nil, err
 			}
 
-			pbBatchReadResult := &pb.BatchReadResult{ReadName: batchCall.ReadName, ReturnVal: encodedRetVal, Error: replyErr}
-			pbBatchGetLatestValuesReply.Results[contractName].Results = append(pbBatchGetLatestValuesReply.Results[contractName].Results, pbBatchReadResult)
+			batchResult.Results = append(batchResult.Results, &pb.BatchReadResult{
+				ReadName:  batchCall.ReadName,
+				ReturnVal: encodedRetVal,
+				Error:     replyErr,
+			})
 		}
 	}
-	return pbBatchGetLatestValuesReply, nil
+
+	return &pb.BatchGetLatestValuesReply{Results: results}, nil
 }
 
 func convertBatchGetLatestValuesRequestToProto(request types.BatchGetLatestValuesRequest, encodeWith EncodingVersion) (*pb.BatchGetLatestValuesRequest, error) {
-	pbRequest := &pb.BatchGetLatestValuesRequest{Requests: make(map[string]*pb.ContractBatch)}
-	for contractName, contractBatch := range request {
-		if _, ok := pbRequest.Requests[contractName]; !ok {
-			pbContractBatch := &pb.ContractBatch{Reads: []*pb.BatchRead{}}
-			pbRequest.Requests[contractName] = pbContractBatch
-		}
+	requests := make([]*pb.ContractBatch, len(request))
```

Comment:
This will render this check `if len(requestContractBatch) != len(resultsContractBatch) ` in ` parseBatchGetLatestValuesReply()` useless